associated-token-account/program/Cargo.toml

```diff
@@ -12,11 +12,11 @@ no-entrypoint = []
 test-sbf = []
 
 [dependencies]
-assert_matches = "1.5.0"
 borsh = "1.5.1"
 num-derive = "0.4"
 num-traits = "0.2"
 solana-program = "2.0.3"
+spl-associated-token-account-client = { version = "1.0.0", path = "../client" }
```

Comment:
Interesting. Is depending on its own client an anti-pattern in the program clients world?

Reply:
In most cases, probably. In a world of generated clients, the instruction creators don't need to exist on the program crate at all, and should be exposed only on the client crate. In this case, however, we want to avoid breaking people that currently get the instruction creators from `spl-associated-token-account`.

For PDA creators, we might have to live with some replicode between the client and program crates.

Reply:
Sounds reasonable, especially for the hand-rolled client here.